### DISCUSSION TITLE
Support mg/m³ unit of measurement for Air Quality sensors

### DISCUSSION THREAD

**Discussion by LexPanov** [home-assistant/architecture#1004] [Entity Models] [answered]
In the discussion under #136, @Jc2k proposed `µg/m³` as the unit supported by "the existing voc sensors."
This unit was implemented later in home-assistant/core#55050

However, I've observed that several Mi Home devices, including in their applications and on-device displays, use `mg/m³` as their unit of measurement. Searching "tVOC air quality sensor" online reveals many examples where this measurement is used also.

For instance, the Xiaomi Qingping Air Monitor, in both its screen display and in the Mi Home, Qingping+, and Qingping IoT apps, shows tVOC measurements in `mg/m³`. This device can be integrated with Home Assistant through the unofficial hass-xiaomi-miot and also via MQTT broker configuration.
Focusing on the latter, the raw values in the MQTT topic are in `ppb`. The Sensirion sensor used in this device aligns with the following conversion information obtained from [Sensirion Gas Sensors FAQ](https://forum.digikey.com/t/sensirion-gas-sensors-faq/5205):

> What is the conversion formula from ppb to mg/m3 for TVOC?
> 1 mg/m3 = 218.77 ppb
> 1 ppb = 0.0045 mg/m3
> The conversion formula assumes a molecular mass of 111.86 g/mol, 25°C and 1.013 bar. The molecular mass is the average molecular mass of the TVOC reference gas mixture.

I applied these guidelines in the value_template transformation in mqtt.sensor as outlined in method detailed in home-assistant/core#104518. However, this led to warning messages in the logs, which was quite disappointing and highlighted the need for a broader acceptance of measurement units.

While an initial online research did not reveal any national standards mandating the use of `mg/m³` for tVOC, several scientific papers, such as [this one](https://doi.org/10.1016/j.buildenv.2014.11.017), utilize `mg/m³`. Although the reasons behind Xiaomi and other manufacturers' choice of `mg/m³` are unclear, the existence of such practices suggests it may be worthwhile to support this unit in the VOC device class for Air Quality sensors.

In discussion thread #858, there was a focus on sorting out measurement units like `µg/m³`, `mg/m³`, `ppm`, and `ppb`. Following this, the "volatile_organic_compounds_parts" class was created, specifically for `ppm` and `ppb` units.

So, currently available are:

-   "volatile_organic_compounds" with `µg/m³`
-   "volatile_organic_compounds_parts" with `ppm` and `ppb`

However, the addition of `mg/m³` as a unit has not yet been implemented.

P.S. I initially submitted a PR at home-assistant/core#104525, but @joostlek kindly requested me to create a discussion here.

**Comment by frenck** [comment]
As with every addition to our core, which integrations are using this?

**Reply by jeeftor** [reply]
Weatherflow's native app uses kg/m3 as its units ... 

<img width="640" alt="image" src="https://github.com/home-assistant/architecture/assets/6491743/c2cde83f-8725-4b23-bfa3-36575ad723e0">

**Reply by frenck** [reply]
Right, but that is not the unit this proposal is about 🤷‍♂️

**Reply by BradleyFord** [reply]
I have a "Aqara TVOC air quality monitor"

It reports in mg/m3

**Comment by matt7aylor** [comment]
Not wishing to hijack the discussion but I think in the specific case of TVOCs discussed here (and others including NOx) the unit that should probably be added is `None` because it is a better description of most sensors (especially the more modern ones which explicitly use a unit-less index).

I have worked in air quality measurement for a number of years (and in laboratory science before that) including testing and building products using many commercially available VOC sensor components. There has been a significant problem in that most (probably all) of the sensor components used in commercially available consumer devices are metal oxides sensors (because they are the only affordable ones for non-specialist devices). Where these devices have been claiming to measure a specific unit for VOCs, they have very little justification in doing so and the measurements are only valuable in a relative sense. [A reasonable discussion of this [here](https://www.forensicsdetectors.com/blogs/articles/voc-detector-air-quality)] (The fact that products claiming to measure the same thing in the same context choose units 1000x different in `µg/m³` and `mg/m³` is a good indication of the lack of measurement quality.)

The industry has started to catch up to this and now we see the new metal oxide sensor chips, such as the [Sensiron SGP40](https://sensirion.com/products/catalog/SGP40), the successor to the SGP3x series referenced above (in [Sensirion Gas Sensors FAQ](https://forum.digikey.com/t/sensirion-gas-sensors-faq/5205) ) or the [Bosch BME68x series](https://www.bosch-sensortec.com/products/environmental-sensors/gas-sensors/bme680/) provide an explicitly unit-less "IAQ index" measurement.

It is maybe not for Home Assistant to ensure that device manufacturers produce honest sensors but to implement support for devices that exist (as is requested here). However, while a proposal for modifications to air quality units is being considered, if any unit is being added to support such air quality sensors it should probably be the option for a `None`/unit-less value as this is what the newly available chips (and the old ones that lie) actually produce.

(I have not had any involvement with Home Assistant development before the last couple of days in attempting to add Air Quality to the matter integration [#108173](https://github.com/home-assistant/core/pull/108173) )

**Reply by matt7aylor** [reply]
For anyone interested there are quite a lot of good resources on the limitations of metal oxide sensors and VOC measurements in the technical downloads section of the [Sensiron SGP41](https://sensirion.com/products/catalog/SGP41) (a metal oxide sensor) product page e.g. [What is Sensiron's VOC index?](https://sensirion.com/media/documents/02232963/6294E043/Info_Note_VOC_Index.pdf)

I guess all I'm saying is that for this particular measurement we might expect to see quite a lot of different (non-convertible) units being used, including no units.

**Comment by frenck** [comment, marked as answer]
Created a PR to resolve this architectural proposal:

https://github.com/home-assistant/core/pull/145325
